.tekton/pipeline-ref.yaml

```diff
@@ -173,6 +178,52 @@ spec:
       workspace: git-auth
     - name: netrc
       workspace: netrc
+  - name: build-front-container
+    params:
+    - name: IMAGE
+      value: "$(params.output-image)-front"
+    - name: DOCKERFILE
+      value: $(params.front-dockerfile)
+    - name: CONTEXT
+      value: $(params.path-context)
+    - name: HERMETIC
+      value: $(params.hermetic)
+    - name: PREFETCH_INPUT
+      value: $(params.prefetch-input)
+    - name: IMAGE_EXPIRES_AFTER
+      value: $(params.image-expires-after)
+    - name: COMMIT_SHA
+      value: $(tasks.clone-repository.results.commit)
+    - name: BUILD_ARGS
+      value:
+      - $(params.build-args[*])
+      - "COMMIT=$(tasks.clone-repository.results.commit)"
+      - "BUILDVERSION=$(params.build-version)"
+      - "DATE=$(tasks.clone-repository.results.commit-timestamp)"
+    - name: BUILD_ARGS_FILE
+      value: $(params.build-args-file)
+    - name: SOURCE_ARTIFACT
+      value: $(tasks.prefetch-dependencies.results.SOURCE_ARTIFACT)
+    - name: CACHI2_ARTIFACT
+      value: $(tasks.prefetch-dependencies.results.CACHI2_ARTIFACT)
+    - name: PLATFORM
+      value: "linux/x86_64"
```

Comment:
if you specify the architecture, won't it reject it when trying to build the full image on a different architecture?

Reply:
Only the build-container task is run multiple times because of the architecture, this is done by using the matrix definition and running the task for each platform value:
https://github.com/netobserv/network-observability-console-plugin/blob/60d35740c525cdf2c7976d1d3180243471249a70/.tekton/pipeline-ref.yaml#L177

The new build-front-container is only run once, the platform here is just to tell konflux which platform to use to build it once.